numba/targets/arrayobj.py

```diff
@@ -1739,6 +1783,307 @@ def _increment_indices_array(context, builder, arrty, arr, indices, end_flag=Non
     _increment_indices(context, builder, arrty.ndim, shape, indices, end_flag)
 
 
+def make_nditer_cls(nditerty):
```

Comment:
Could this be perhaps named `make_nditer_struct` or similar? I have the same objection to `make_array` that IIRC doesn't create an array but creates the struct necessary for wiring up an array?